packages/flutter/test/material/text_field_test.dart

```diff
@@ -3341,4 +3341,49 @@ void main() {
     expect(minOffset, 0.0);
     expect(maxOffset, 50.0);
   });
+
+  testWidgets('TextField style is merged with theme', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/23994
+
+    final ThemeData themeData = ThemeData(
+      textTheme: TextTheme(
+        subhead: TextStyle(
+          color: Colors.blue[500],
+        ),
+      ),
+    );
+
+    Widget buildFrame(TextStyle style) {
+      return MaterialApp(
+        theme: themeData,
+        home: Material(
+          child: Center(
+            child: TextField(
+              style: style,
+            ),
+          ),
+        ),
+      );
+    }
+
+    // Empty TextStyle is overridden by theme
+    await tester.pumpWidget(buildFrame(TextStyle()));
+    EditableText editableText = tester.widget(find.byType(EditableText));
+    expect(editableText.style.color, themeData.textTheme.subhead.color);
+
+    // Properties set on TextStyle override theme
+    const Color setColor = Colors.red;
+    await tester.pumpWidget(buildFrame(const TextStyle(color: setColor)));
+    editableText = tester.widget(find.byType(EditableText));
+    expect(editableText.style.color, setColor);
+
+    // inherit: false causes nothing to be merged in from theme
+    await tester.pumpWidget(buildFrame(TextStyle(
+      fontSize: 24.0,
+      textBaseline: TextBaseline.alphabetic,
```

Comment:
@HansMuller I found that I got errors when I used a TextStyle with `inherited: false` and didn't also provide these two parameters `fontSize` and `textBaseline`.  I'm sure they're being merged onto this somewhere internally, but `inherited: false` prevents it.  Is that anything to be concerned about?

Reply:
Maybe related to this crash? https://github.com/flutter/flutter/issues/8728

That crash doesn't happen for me (with `inherited: true`), but does look similar to this.